tests/tests_tasks.py

```diff
@@ -87,25 +99,40 @@ def test_run_task(self, post_callback):
-        self.assertEqual(self.response_data['results']['response_time']['highest'], 123.0)
-        self.assertEqual(self.response_data['results']['response_time']['lowest'], 123.0)
-
-        self.assertEqual(self.response_data['results']['backend_metrics']['memc_dupes']['count'], 10)
-        self.assertEqual(self.response_data['results']['backend_metrics']['memc_dupes']['highest'], 4.0)
-        self.assertEqual(self.response_data['results']['backend_metrics']['memc_misses']['count'], 10)
-        self.assertEqual(self.response_data['results']['backend_metrics']['memc_misses']['highest'], 1.0)
-        self.assertEqual(self.response_data['results']['backend_metrics']['query_master']['highest'], 314.0)
-        self.assertEqual(self.response_data['results']['backend_metrics']['query_slave']['highest'], 11.0)
-        self.assertEqual(self.response_data['results']['backend_metrics']['query_time']['highest'], 0.363579)
-        self.assertEqual(self.response_data['results']['backend_metrics']['server_time']['highest'], 1.869824)
-
-        self.assertEqual(self.response_data['results']['phantomas_metrics']['js_size']['highest'], 927221.0)
-        self.assertEqual(self.response_data['results']['phantomas_metrics']['content_length']['highest'], 1922742.0)
-        self.assertEqual(self.response_data['results']['phantomas_metrics']['css_count']['highest'], 5.0)
-        self.assertEqual(self.response_data['results']['phantomas_metrics']['body_size']['highest'], 1304562.0)
-        self.assertEqual(self.response_data['results']['phantomas_metrics']['html_size']['highest'], 124541.0)
-        self.assertEqual(self.response_data['results']['phantomas_metrics']['other_count']['highest'], 19.0)
+        results = self.response_data['results']
+
+        response_times = get_stats_from_requests('requests', 'server.app.response_time')
+        self.assertEqual(response_times.count, 10)
+        self.assertEqual(response_times.min, 123.0)
+        self.assertEqual(response_times.max, 123.0)
+
+        memcached_dupes = get_stats_from_requests('mw_profiler', 'server.app.memcached.dupe_count')
+        memcached_misses = get_stats_from_requests('mw_profiler', 'server.app.memcached.miss_count')
+        queries_master = get_stats_from_requests('mw_profiler', 'server.app.database.queries.master_count')
+        queries_slave = get_stats_from_requests('mw_profiler', 'server.app.database.queries.slave_count')
+        queries_time = get_stats_from_requests('mw_profiler', 'server.app.database.queries.time')
+        response_times = get_stats_from_requests('mw_profiler', 'server.app.response_time')
+        self.assertEqual(memcached_dupes.count, 10)
+        self.assertEqual(memcached_dupes.max, 4.0)
+        self.assertEqual(memcached_misses.count, 10)
+        self.assertEqual(memcached_misses.max, 1.0)
+        self.assertEqual(queries_master.max, 314.0)
+        self.assertEqual(queries_slave.max, 11.0)
+        self.assertEqual(queries_time.max, 0.363579)
```

Comment:
Be aware that due to float representation it may not be a true. I had to change it to assertAlmostEqual() when passing this value through multiple json decode/encode functions... 
see: https://github.com/Wikia/sparrow/pull/30